modules/frontend/combiner/trace_by_id_v2.go

```diff
@@ -21,8 +23,13 @@ func NewTraceByIDV2(maxBytes int, marshalingFormat string) Combiner {
 			// dedupe duplicate span ids
 			deduper := newDeduper()
 			traceResult = deduper.dedupe(traceResult)
-
 			resp.Trace = traceResult
+
+			if combiner.IsPartialTrace() {
```

Comment:
[Below](https://github.com/grafana/tempo/pull/3941/files#diff-a1fec687cd87c7167aee355f831ebae6beb368066e3bfbf092ef0aaf5e216aa2R317-R320) we are communicating from the querier to the frontend that there is a partial trace, but we are not handling it here.

The rest of this PR is looking good. My final concern is that we should handle `partial.Status == tempopb.TraceByIDResponse_PARTIAL` at the top of this `combine` func.

Reply:
My rationale was that a partial trace will always have a size bigger than the maxBytes so it won't be needed more changes.

If we want to be more performance-wise we can inspect the status and don't do any kind of processing

Reply:
It does seem odd to me to pass back "partial trace" from the queriers and not honor it here. We should still combine the results here exactly like we're doing. It's possible (although unlikely) to calculate a partial trace in a querier and not come to the same conclusion in the frontend. For completeness if a querier tells sends back "partial trace" we should propagate it back to the caller.